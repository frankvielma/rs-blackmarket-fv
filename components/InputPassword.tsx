import Image from 'next/image'
import { useState } from 'react'

export default function InputPassword({
  id = '',
  label = '',
  placeholder = '',
}) {
  const [isActive, setActive] = useState(true)
  const toggleVisibility = () => {
    setActive(!isActive)
  }

  return (
    <div className="relative my-4">
      <label htmlFor="password" className="text-sm md:text-base">
        {label}
      </label>
      <div className="flex flex-col">
        <input
          className="custom-focus h-[34px] w-auto rounded-md border border-dark-violet ps-3 placeholder:text-base placeholder:text-dark-gray hover:border-hover active:outline-none active:ring-2 active:ring-offset-2 md:h-[44px]"
          id={id}
          placeholder={placeholder}
          type={isActive ? 'password' : 'text'}
        />

        <button
          className="custom-focus absolute right-4 mt-2 md:mt-4"
          type="button"
          onClick={toggleVisibility}
        >
          <Image
            src={`/icons/visibility_${isActive ? 'off' : 'on'}.svg`}
            height={0}
            width={0}
            sizes="100vw"
            className="w-4 h-auto"
            alt=""
          />
        </button>
      </div>
    </div>
  )
}
