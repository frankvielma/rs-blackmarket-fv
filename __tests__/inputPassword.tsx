import { render, fireEvent, screen } from '@testing-library/react'
import InputPassword from '@/components/InputPassword'

describe('InputPassword', () => {
  it('renders InputPassword unchanged', () => {
    const { container } = render(<InputPassword />)
    expect(container).toMatchSnapshot()
  }),
    it('calls the onChange function', () => {
      render(<InputPassword label={'Password'} type={'password'} id={'password'} />)
      const inputElement = screen.getByLabelText('Password')
      fireEvent.change(inputElement, { target: { value: 'test value' } })
      expect((inputElement as HTMLInputElement).value).toBe('test value')
    }),
    it('calls the handleBlur function', () => {
      render(<InputPassword label={'Password'} type={'password'} id={'password'} />)
      const inputElement = screen.getByLabelText('Password')
      fireEvent.blur(inputElement)
      expect(inputElement).toHaveClass(
        'border border-red-600 custom-focus h-[34px] w-auto rounded-md border border-dark-violet ps-3 placeholder:text-base placeholder:text-dark-gray hover:border-hover active:outline-none active:ring-2 active:ring-offset-2 md:h-[44px]'
      )
    })
})
