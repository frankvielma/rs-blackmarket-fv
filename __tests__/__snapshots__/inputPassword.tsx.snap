// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputPassword renders InputPassword unchanged 1`] = `
<div>
  <div
    class="relative my-4"
  >
    <label
      class="text-sm md:text-base"
      for=""
    />
    <div
      class="flex flex-col"
    >
      <input
        class="border custom-focus h-[34px] w-auto rounded-md border border-dark-violet ps-3 placeholder:text-base placeholder:text-dark-gray hover:border-hover active:outline-none active:ring-2 active:ring-offset-2 md:h-[44px]"
        id=""
        pattern=".{8,}"
        placeholder=""
        required=""
        type="password"
      />
      <span />
      <button
        class="custom-focus absolute right-4 mt-2 md:mt-4"
        type="button"
      >
        <span
          class="sr-only"
        >
          Show password
        </span>
        <img
          alt=""
          class="h-auto w-4"
          data-nimg="1"
          decoding="async"
          height="0"
          loading="lazy"
          src="/icons/visibility_off.svg"
          style="color: transparent;"
          width="0"
        />
      </button>
    </div>
  </div>
</div>
`;
